runtime/v2/sandbox.go

```diff
@@ -0,0 +1,254 @@
+		SandboxID:  sandboxID,
+		BundlePath: bundle.Path,
+	}); err != nil {
+		return nil, errors.Wrap(err, "failed to start sandbox instance")
+	}
+
+	if err := m.sandboxes.Add(ctx, shim); err != nil {
+		return nil, err
+	}
+
+	return shim, nil
+}
+
+// Shutdown will turn down existing sandbox instance
+func (m *SandboxManager) Shutdown(ctx context.Context, sandboxID string) error {
+	shim, err := m.getShim(ctx, sandboxID)
+	if err != nil {
+		return err
+	}
+
+	svc, err := shim.Sandbox(ctx)
+	if err != nil {
+		return err
+	}
+
+	if _, err := svc.StopSandbox(ctx, &proto.StopSandboxRequest{
+		SandboxID: sandboxID,
+	}); err != nil {
+		return errors.Wrapf(err, "failed to stop %s", sandboxID)
+	}
+
+	m.tasks.deleteShim(shim)
+	return nil
+}
+
+// Pause will freeze running sandbox instance
+func (m *SandboxManager) Pause(ctx context.Context, sandboxID string) error {
+	svc, err := m.getSvc(ctx, sandboxID)
+	if err != nil {
+		return err
+	}
+
+	if _, err := svc.PauseSandbox(ctx, &proto.PauseSandboxRequest{
+		SandboxID: sandboxID,
+	}); err != nil {
+		return errors.Wrapf(err, "failed to pause sandbox %s", sandboxID)
+	}
+
+	return nil
+}
+
+// Resume will unfreeze previously paused sandbox instance
+func (m *SandboxManager) Resume(ctx context.Context, sandboxID string) error {
+	svc, err := m.getSvc(ctx, sandboxID)
+	if err != nil {
+		return err
+	}
+
+	if _, err := svc.ResumeSandbox(ctx, &proto.ResumeSandboxRequest{
+		SandboxID: sandboxID,
+	}); err != nil {
+		return errors.Wrapf(err, "failed to resume sandbox %s", sandboxID)
+	}
+
+	return nil
+}
+
+// Ping is a lightweight shim call to check whether sandbox alive
+func (m *SandboxManager) Ping(ctx context.Context, sandboxID string) error {
+	svc, err := m.getSvc(ctx, sandboxID)
+	if err != nil {
+		return err
+	}
+
+	if _, err := svc.PingSandbox(ctx, &proto.PingRequest{
+		SandboxID: sandboxID,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Status will return current status of the running sandbox instance
+func (m *SandboxManager) Status(ctx context.Context, sandboxID string) (*types.Any, error) {
+	svc, err := m.getSvc(ctx, sandboxID)
+	if err != nil {
+		return nil, err
+	}
+
+	status, err := svc.SandboxStatus(ctx, &proto.SandboxStatusRequest{
+		SandboxID: sandboxID,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to query sandbox %s status", sandboxID)
+	}
+
+	return status.Status, nil
+}
+
+// ID of the sandbox runtime
+func (m *SandboxManager) ID() string {
+	return fmt.Sprintf("%s.%s", plugin.RuntimePluginV2, "sandbox")
+}
+
+// Create creates a task with the provided id and options.
+func (m *SandboxManager) Create(ctx context.Context, id string, opts runtime.CreateOpts) (runtime.Task, error) {
+	container, err := m.tasks.containers.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	// No sandbox, fallback to task manager
+	if container.SandboxID == "" {
+		return m.tasks.Create(ctx, id, opts)
+	}
+
```

Comment:
It may never be implemented here. Since `containerToProto/containerFromProto` will lose the SandboxID field. This causes the containerStore to not store the SandboxID.